src/main/java/com/mraof/minestuck/entity/ai/attack/AnimatedAttackWhenInRangeGoal.java

```diff
@@ -110,6 +117,18 @@ protected boolean withinRange(@Nonnull LivingEntity target)
 		return belowMaximumRange(target) && aboveMinimumRange(target);
 	}
 	
+	protected boolean withinAttackCone(@Nonnull LivingEntity target)
+	{
+		Vec3 attackDirection = Vec3.directionFromRotation(0, this.entity.getVisualRotationYInDegrees() + this.attackDirectionOffset);
+		Vec3 targetDirection = this.entity.position().vectorTo(target.position()).normalize();
+		targetDirection = new Vec3(targetDirection.x, 0, targetDirection.z);
+		
+		float angleOfTargetFromConeCenter = (180.0F / 2.0F) * ((((float)attackDirection.dot(targetDirection)) * -1.0F) + 1.0F);
+		
+		return angleOfTargetFromConeCenter <= this.attackConeAngle;
+		
+	}
```

Comment:
The method name is pretty self descriptive(you did a good job!) but mind adding a comment to explain a little bit of the math to help anyone who might try adding new attacks?